ruby/ext/google/protobuf_c/encode_decode.c

```diff
@@ -1103,6 +1253,12 @@ VALUE Message_encode(VALUE klass, VALUE msg_rb) {
   VALUE descriptor = rb_ivar_get(klass, descriptor_instancevar_interned);
   Descriptor* desc = ruby_to_Descriptor(descriptor);
 
+  if (desc->has_udts_transitively) {
```

Comment:
Likewise, can't we do this inline with the actual put*() functions? Is that complex in some way that I am missing?